src/python/diffie-hellman/diffie-hellman.md

```diff
@@ -0,0 +1,225 @@
+måte, og det har Python innebygd. I dette eksempelet kalles 12
+*modulusen*.
+
+Åpne IDLE, men uten å lage en ny fil. Der det står `>>>` kan vi skrive
+inn kode for å se hva den gjør.
+
+## Sjekkliste {.check}
+
+- [ ] Skriv `(11 + 4) % 12` og trykk Enter. Du skal da få svaret 3. Om
+      du sammenligner med det vi gjorde i primtallsleksjonen, ser du
+      kanskje at 15 delt på 12 gir 3 til rest.
+- [ ] Det fungerer også med multiplikasjon. Skriv `(5 * 4) % 12` og
+      trykk Enter. Da skal du få 8, siden 5 ⋅ 4 = 20 = 12 + 8.
+- [ ] Prøv deg fram med litt flere eksempler selv. Prøv å forstå hva du
+      skal få før du faktisk ber Python regne det ut for deg.
+- [ ] Regn ut `(3 * 4) % 12`. Hva får du?
+- [ ] Vi kan også regne ut potenser. I matematikkboka skrives det som
+      4<sup>3</sup>, mens tilsvarende Python-syntaks er `4**3`. Bruk
+      IDLE til å regne ut `4**3 % 12` og `6**2 % 12`.
+
+# Steg 2: Finne en nøkkel for Cæsar-chifferet {.activity}
+
+I denne aktiviteten er det ment at du skal samarbeide med noen andre, så
+vi skal strukturere koden vår i funksjoner som du kan bruke fra
+IDLE-konsollen. Etter hvert skal vi utvide koden vår slik at den i
+prinsippet kan brukes til alle formål.
+
+## Sjekkliste {.check}
+
+- [ ] Åpne en ny fil i IDLE, og skriv inn den følgende koden
+  ```python
+  p = 23
+  g = 5
+
+  def generate_share(a, g, p):
+      return g**a % p
+
+  def generate_secret(other_share, a, p):
+      return other_share**a % p
+  ```
+- [ ] Lagre filen og kjør modulen ved å trykke F5. Du får da opp et
+      Python-skall der du kan skrive kommandoer selv. Finn på et
+      tilfeldig tall mellom 1 og 23, og skriv `generate_share(<ditt tall>, g, p)`.
+      Da får du ut et nytt tall. Dette er det offentlige tallet du skal
+      si høyt til vennen til på motsatt side av rommet. Han eller hun
+      skal gjøre det samme.
+- [ ] Skriv `generate_secret(<tallet vennen din ropte>, <ditt tall>, p)`,
+      og trykk Enter. Det tallet du får ut nå er den hemmelige nøkkelen
+      som du og vennen din deler, men som ingen andre kan vite om.
+- [ ] Finn fram Cæsar-chifferet, og bruk den hemmelige nøkkelen som
+      `secret`. Send en melding til vennen din, og se om han eller hun
+      får til å dekryptere korrekt.
+
+Dersom du jobber alene, kan du teste dette ved å kjøre `generate_share`
+og `generate_secret` to ganger. Da må du passe på å bruke riktig tall på
+riktig plass.
+
+# Steg 3: Bedre tilfeldighet {.activity}
+
+I forrige steg ble du bedt om å skrive inn et tilfeldig tall. Mennesker
+er ikke flinke til å finne på tilfeldige tall, så det er på tide å dra
+inn noe av det vi har lært fra før. Se over
+[Tilfeldige tall](../tilfeldige_tall/tilfeldige_tall.html) på nytt
+dersom du har glemt hva vi gjorde der.
+
+## Sjekkliste {.check}
+
+- [ ] Endre koden din slik at den nå ser slik ut.
+  ```python
+  from secrets import randbelow
+
+  p = 23
+  g = 5
+
+  def generate_share(g, p):
+      a = randbelow(p)
```

Comment:
tror du må definere a utenom også dersom du skal lagre verdien for bruk i generate_secret også, med mindre du mener at de skal huske/skrive inn a selv

Reply:
$a$ returneres av koden, så da kan de ta notat av den.